flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -254,13 +262,14 @@ public ExecutionGraph(
 			ExecutionContext executionContext,
 			JobID jobId,
 			String jobName,
+			JobType jobType,
 			Configuration jobConfig,
 			FiniteDuration timeout,
 			List<BlobKey> requiredJarFiles,
 			List<URL> requiredClasspaths,
 			ClassLoader userClassLoader) {
 
-		if (executionContext == null || jobId == null || jobName == null || jobConfig == null || userClassLoader == null) {
+		if (executionContext == null || jobId == null || jobName == null || jobType == null || jobConfig == null || userClassLoader == null) {
 			throw new NullPointerException();
```

Comment:
I think it would be good to know which of the fields was actually `null`. We could use

```
executionContext = Preconditions.checkNotNull(executionContext);
...
```